torch/csrc/distributed/rpc/request_callback_no_python.cpp

```diff
@@ -216,18 +256,306 @@ bool RequestCallbackNoPython::processScriptRemoteCallOp(
   return false;
 }
 
+void RequestCallbackNoPython::processScriptRRefFetchCall(
+    RpcCommandBase& rpc,
+    const std::function<void(Message)>& markComplete,
+    const int64_t messageId,
+    const std::shared_ptr<FutureMessage>& responseFuture) const {
+  auto& srf = static_cast<ScriptRRefFetchCall&>(rpc);
+  auto& ctx = RRefContext::getInstance();
+
+  auto futureOwner = ctx.getOwnerRRef(srf.rrefId());
+
+  if (futureOwner->completed()) { // optional fast-path
+    // the OwnerRRef has been created
+    const auto& rref = futureOwner->constValue();
+    if (rref->hasValue()) {
+      markComplete(ScriptRRefFetchRet({rref->getValue()}).toMessage());
+      return;
+    }
+  }
+
+  futureOwner->addCallback([responseFuture, messageId, futureOwner]() {
+    const auto& rref = futureOwner->constValue();
+    auto whenValueSet = rref->getFuture();
+
+    // Our response is satisfied when the rpc.remote() request
+    // finishes executing on the owner.
+    whenValueSet->addCallback(
+        [responseFuture, messageId, rref, whenValueSet]() {
+          if (whenValueSet->hasError()) {
+            responseFuture->setError(
+                whenValueSet->tryRetrieveErrorMessage());
+            return;
+          }
+          try {
+            Message m = ScriptRRefFetchRet({rref->getValue()}).toMessage();
+            m.setId(messageId);
+            responseFuture->markCompleted(std::move(m));
+          } catch (const std::exception& e) {
+            responseFuture->setError(e.what());
+          }
+        });
+  });
+}
+
 void RequestCallbackNoPython::processPythonRRefFetchCall(
     RpcCommandBase& rpc,
     const int64_t messageId,
     const std::shared_ptr<FutureMessage>& responseFuture) const {
   C10_THROW_ERROR(Error, "Python call not supported!");
 }
 
+void RequestCallbackNoPython::processRRefUserDelete(RpcCommandBase& rpc,
```

Comment:
clang-format might give the following result:

```
void RequestCallbackNoPython::processRRefUserDelete(
    RpcCommandBase& rpc,
    const std::function<void(Message)>& markComplete) const {
```

Reply:
Thanks! I need to set up the clang-format git hook so that it runs before I commit. Is there a reason the CI checks for clang-format or clang-tidy don't throw any errors?